### ISSUE TITLE
Moving contexts between threads.

### ISSUE THREAD

**Issue by jwilm** [tomaka/glutin#784]
I've been trying to find a way with Glutin to separate input polling/waiting from the render thread. With GLFW, there is a way to [use the render context on a separate thread](https://github.com/PistonDevelopers/glfw-rs/blob/master/examples/render_task.rs) which would achieve  my goal. Is this possible with current Glutin APIs?

**Comment by tomaka** [comment]
Glutin also provides `make_current`.
What's lacking is the possibility to call `make_current(null)` at the end, but you don't need it unless you want to change your rendering thread from time to time.

**Comment by jwilm** [comment]
Is Window actually `Send` + `Sync`? The methods are all `&self`, and I don't get any errors sending to another thread or using it in parallel from two threads. It is a bit surprising since that's been uncommon in these sorts of libraries historically, and the fact that `WindowProxy` exists also suggests that shouldn't be true of `Window`.

**Comment by tomaka** [comment]
Window can be Send+Sync on Windows (the OS) and Linux, but not OSX.

I don't know OSX well enough and I have yet to get an answer as to whether it's possible to fix this in a hacky way.

**Comment by fkaa** [comment]
On OSX the context can be Send+Sync, not the Window.

**Comment by jwilm** [comment]
I haven't looked at the Mac implementation at all, but missing `make_current(null)` doesn't seem to be a problem there. Must not be glx on that platform? There have been no issues there polling for input from one thread (as long as it's the main thread) and using the context from a secondary thread.

On Linux with X and glx, `make_current(null)` or `clear_context()` would be nice to have.

**Comment by ZeGentzy** [comment]
Merged with #1129
